libraries/grpc-sdk/src/utilities/normalizeParams.ts

```diff
@@ -0,0 +1,10 @@
+export function normalizeParams(givenParams: any[], fnParamKeys: string[]) {
```

Comment:
`givenParams`'s type structure feels ambiguous and somewhat error-prone.
Object variant shouldn't be array-nested (nor should the actual function type unions wrap the object subtypes in an array).

This makes it possible for an array of objects to be passed along instead of a plain object value.
This skews ergonomics, in that you'd need to array-wrap the actual values during the call, but more importantly introduces an unchecked flow.
What happens if an array of object-formatted args is passed in?